File: StockManagementAPI/StockManagementAPI/Vehicle.cs
namespace StockManagementAPI
{
  public class Vehicle
  {
    public int Id { get; set; }
    public string RegistrationNumber { get; set; } = string.Empty;
    public string Manufacturer { get; set; } = string.Empty;
    public string ModelDescription { get; set; } = string.Empty;
    public int ModelYear { get; set; }
    public int KilometreReading { get; set; } = 0;
    public string Colour { get; set; } = string.Empty;
    public string VIN { get; set; } = string.Empty;
    public string Accessories { get; set; } = string.Empty;
    public int RetailPrice  { get; set; } = 0;
    public int CostPrice { get; set; } = 0;
    public DateTime DTCreated { get; set; }
    public DateTime DTUpdated { get; set; }

  }
}

File: src/app/model/vehicle.ts
import { Bytes } from "@angular/fire/firestore";
import { Accessory } from "./accessory";

export class Vehicle {
    id?: number = 0;
    modelYear : number = 0;
    modelDescription ='';
    manufacturer ='';
    colour ='';
    registrationNumber='';
    vin ='';
    kilometreReading : number = 0;
    retailPrice : number = 0;
    costPrice  : number = 0;
    imagesArray: string[]=[];
    accessories = '';
    dtCreated = '';
    dtUpdated = '';
}
